larray_editor/mapping.py

```diff
@@ -0,0 +1,222 @@
+
+    def _map_global_to_filtered(self, k):
+        """
+        map global (unfiltered) ND key to local (filtered) 2D key
+
+        Parameters
+        ----------
+        k: tuple
+            Labels associated with the modified element of the non-filtered array.
+
+        Returns
+        -------
+        tuple
+            Positional index (row, column) of the modified data cell.
+        """
+        assert isinstance(k, tuple) and len(k) == self.la_data.ndim
+        dkey = {axis_id: axis_key for axis_key, axis_id in zip(k, self.la_data.axes.ids)}
+        # transform global dictionary key to "local" (filtered) key by removing
+        # the parts of the key which are redundant with the filter
+        for axis_id, axis_filter in self.current_filter.items():
+            axis_key = dkey[axis_id]
+            if np.isscalar(axis_filter) and axis_key == axis_filter:
+                del dkey[axis_id]
+            elif not np.isscalar(axis_filter) and axis_key in axis_filter:
+                pass
+            else:
+                # that key is invalid for/outside the current filter
+                return None
+        # transform (axis:label) dict key to positional ND key
+        try:
+            index_key = self.filtered_data._translated_key(dkey)
+        except ValueError:
+            return None
+        # transform positional ND key to positional 2D key
+        strides = np.append(1, np.cumprod(self.filtered_data.shape[1:-1][::-1], dtype=int))[::-1]
+        return (index_key[:-1] * strides).sum(), index_key[-1]
+
+    def change_filter(self, axis, indices):
+        # must be done before changing self.current_filter
+        self.update_changes()
+        axis_id = self.la_data.axes.axis_id(axis)
+        if not indices or len(indices) == len(axis.labels):
+            if axis_id in self.current_filter:
+                del self.current_filter[axis_id]
+        else:
+            if len(indices) == 1:
+                self.current_filter[axis_id] = axis.labels[indices[0]]
+            else:
+                self.current_filter[axis_id] = axis.labels[indices]
+        self.update_filtered_data()
+        return self.filtered_data
+
+    def clear_changes(self):
+        self.changes.clear()
+        self.data_model.changes.clear()
+
+    def accept_changes(self):
+        """Accept changes"""
+        # update changes
+        self.update_changes()
+        # update internal data
+        axes = self.la_data.axes
+        for k, v in self.changes.items():
+            self.la_data.i[axes.translate_full_key(k)] = v
+        # update models
+        self.update_filtered_data()
+        # clear changes
+        self.clear_changes()
+        # return modified data
+        return self.la_data
+
+    def reject_changes(self):
+        """Reject changes"""
+        # clear changes
+        self.clear_changes()
+        self.data_model.reset_minmax()
+        self.data_model.reset()
```

Comment:
accept_changes returns the data here because it used to update view (nb decimals, background color, ...). See accept_changes from ArrayEditorWidget.